sql/src/main/java/io/crate/metadata/pgcatalog/PgClassTable.java

```diff
@@ -22,107 +22,93 @@
-            .register("reloftype", INTEGER, () -> constant(0))
-            .register("relowner", INTEGER, () -> constant(0))
-            .register("relam", INTEGER, () -> constant(0))
-            .register("relfilenode", INTEGER, () -> constant(0))
-            .register("reltablespace", INTEGER, () -> constant(0))
-            .register("relpages", INTEGER, () -> constant(0))
-            .register("reltuples", FLOAT, () -> forFunction(r -> (float) tableStats.numDocs(r.ident())))
-            .register("relallvisible", INTEGER, () -> constant(0))
-            .register("reltoastrelid", INTEGER, () -> constant(0))
-            .register("relhasindex", BOOLEAN, () -> constant(false))
-            .register("relisshared", BOOLEAN, () -> constant(false))
-            .register("relpersistence", STRING, () -> constant(PERSISTENCE_PERMANENT))
-            .register("relkind", STRING, () -> forFunction(r -> r.relationType() == RelationType.VIEW ? KIND_VIEW : KIND_TABLE))
-            .register("relnatts", SHORT, () -> forFunction(r -> (short) r.columns().size()))
-            .register("relchecks", SHORT, () -> constant((short) 0))
-            .register("relhasoids", BOOLEAN, () -> constant(false))
-            .register("relhaspkey", BOOLEAN, () -> forFunction(r -> r.primaryKey().size() > 0))
-            .register("relhasrules", BOOLEAN, () -> constant(false))
-            .register("relhastriggers", BOOLEAN, () -> constant(false))
-            .register("relhassubclass", BOOLEAN, () -> constant(false))
-            .register("relrowsecurity", BOOLEAN, () -> constant(false))
-            .register("relforcerowsecurity", BOOLEAN, () -> constant(false))
-            .register("relispopulated", BOOLEAN, () -> constant(true))
-            .register("relreplident", STRING, () -> constant("p"))
-            .register("relispartition", BOOLEAN, () -> constant(false))
-            .register("relfrozenxid", INTEGER,() -> constant(0))
-            .register("relminmxid", INTEGER, () -> constant(0))
-            .register("relacl", new ArrayType<>(ObjectType.untyped()), () -> constant(null))
-            .register("reloptions", STRING_ARRAY, () -> constant(null))
-            .register("relpartbound", ObjectType.untyped(), () -> constant(null));
+    public static SystemTable<Entry> create(TableStats tableStats) {
+        return SystemTable.<Entry>builder()
+            .add("oid", INTEGER, x -> x.oid)
+            .add("relname", STRING, x -> x.name)
+            .add("relnamespace", INTEGER, x -> x.schemaOid)
+            .add("reltype", INTEGER, x -> 0)
+            .add("reloftype", INTEGER, x -> 0)
+            .add("relowner", INTEGER, x -> 0)
+            .add("relam", INTEGER, x -> 0)
+            .add("relfilenode", INTEGER, x -> 0)
+            .add("reltablespace", INTEGER, x -> 0)
+            .add("relpages", INTEGER, x -> 0)
+            .add("reltuples", FLOAT, x -> (float) tableStats.numDocs(x.ident))
```

Comment:
`Primary keys` should actually not have `reltuples` value, not relevant for the kafka usecase.

Reply:
Could we use  `relkind` / `relationType` to avoid the `numDocs` call then?

Reply:
it is addressed now.